addons/l10n_no/models/account_move.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from stdnum import luhn
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.move"
+
+    def _get_invoice_reference_no_invoice(self):
+        """ This computes the reference based on the Odoo format.
+            We calculat reference using invoice number and
+            partner id and added control digit at last.
+        """
+        return self._get_kid_number()
+
+    def _get_invoice_reference_no_partner(self):
+        """ This computes the reference based on the Odoo format.
+            We calculat reference using invoice number and
+            partner id and added control digit at last.
+        """
+        return self._get_kid_number()
+
+    def _get_kid_number(self):
+        self.ensure_one()
+        invoice_name = ''.join([i for i in self.name if i.isdigit()]).zfill(7)
+        ref = (str(self.partner_id.id).zfill(7)[-7:] + invoice_name[-7:])
+        return luhn.validate(ref + luhn.calc_check_digit(ref))
```

Comment:
Is the validate necessary or just a precaution?

Reply:
@hpr-odoo

Reply:
@jco-odoo I think validate is unnecessary here because we will always going to have a valid KID number thanks to <a href="https://arthurdejong.org/python-stdnum/doc/1.11/index.html#stdnum.module.calc_check_digit">`calc_check_digit`</a>, also renamed the class

Reply:
Do you leave it out then?  (it would only bring confusion)

Reply:
@rgo-odoo

Reply:
@jco-odoo I removed validate() call and rebased the branch